pkg/operator/etcdmemberlistercontroller/etcdmemberlistercontroller.go

```diff
@@ -0,0 +1,168 @@
+package etcdmemberlistercontroller
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"sync"
+	"time"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+	errorsutil "k8s.io/apimachinery/pkg/util/errors"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cluster-etcd-operator/pkg/dnshelpers"
+	"github.com/openshift/cluster-etcd-operator/pkg/etcdcli"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+var (
+	errNilMembersCache = errors.New("member cache is nil")
+	errKeyExist        = errors.New("key already exist")
+	errKeyNotExist     = errors.New("key does not exist")
+)
+
+type membersCache struct {
```

Comment:
@tjungblu  would this make sense ? .. I am gonna embed this within the controller

Reply:
I think the struct makes sense, I'm not sure why you wanted to have it as a dedicated controller (or maybe I did).

it's more sensible to me to put this into `etcdClientGetter` struct in the etcdcli, then `MemberList` can always return a cached response or fetch when outdated.

Reply:
something along those lines:

```
package etcdcli

import (
	"context"
	"go.etcd.io/etcd/api/v3/etcdserverpb"
)

type CachedMembers struct {
	// some caching logic
}

func (c CachedMembers) MemberList(ctx context.Context) ([]*etcdserverpb.Member, error) {
	//TODO implement me
	panic("implement me")
}

func (c CachedMembers) VotingMemberList(ctx context.Context) ([]*etcdserverpb.Member, error) {
	//TODO implement me
	panic("implement me")
}

func (c CachedMembers) HealthyMembers(ctx context.Context) ([]*etcdserverpb.Member, error) {
	//TODO implement me
	panic("implement me")
}

func (c CachedMembers) HealthyVotingMembers(ctx context.Context) ([]*etcdserverpb.Member, error) {
	//TODO implement me
	panic("implement me")
}

func (c CachedMembers) UnhealthyMembers(ctx context.Context) ([]*etcdserverpb.Member, error) {
	//TODO implement me
	panic("implement me")
}

func (c CachedMembers) UnhealthyVotingMembers(ctx context.Context) ([]*etcdserverpb.Member, error) {
	//TODO implement me
	panic("implement me")
}

```